flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/JobManagerWatermarkTracker.java

```diff
@@ -129,6 +144,11 @@ public Map<String, WatermarkState> add(
             } catch (Exception e) {
                 throw new RuntimeException(e);
             }
+            // no op to get global watermark without updating it
+            if (value.noOp) {
+                addCount--;
+                return accumulator;
+            }
             WatermarkState ws = accumulator.get(value.id);
             if (ws == null) {
                 accumulator.put(value.id, ws = new WatermarkState());
```

Comment:
Should we also reset the WatermarkState for the idle subtask (inside `if (value.noOp)` )?

Reply:
This would reset the state for the local watermark of the subtask to WatermarkState's default watermark which is Long.MIN_VALUE. I think it's best to leave its local watermark as is, especially since the local watermark does not change when idle.